force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,10 +170,38 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        //list of opps that have the same name as oppnames if any (need to check if empty or already exist)
+        List<Opportunity> oppList = new List<Opportunity>([SELECT Id, Name FROM Opportunity WHERE Name = :oppNames]); 
+        //find acct, limit one
+        List<Account> accountsList = new List <Account>([SELECT Id FROM Account WHERE Name = :accountName LIMIT 1]);
+        //create instance of an acct object?
+        Account singleAcct = new Account();
+
+        //if the list is not empty ie if it returns a result, save the whole record into an object variable?
+        if (!accountsList.isEmpty()){
+            singleAcct.Id = accountsList.get(0).Id;
+        
+            } else {
+            singleAcct.Name = accountName;
+            insert singleAcct;
+        }
+        //check if opplist is empty if yes create opps
+        if (oppList.isEmpty()) {
+            //loop through the oppnames. set acct id = above acct id
```

Comment:
This means that your update below only happen when you dont find any opportunities at all. What is there are some opportunities that are found but not all?